### ISSUE TITLE
Kops for Openstack

### ISSUE THREAD

**Issue by drekle** [kubernetes/kops#6310]
**1. Describe IN DETAIL the feature/behavior/change you would like to see.**
Addition of all openstack model elements and tasks to instantiate an openstack cloud.
Protokube for Openstack
Nodeup changes for Openstack

**2. Feel free to provide a design supporting your feature request.**
https://github.com/kubernetes/kops/pull/6351

https://github.com/kubernetes/kops/issues/3566

**Comment by drekle** [comment]
I will reopen my PR after I have tested my changes.

**Comment by drekle** [comment]
Closing in favor of cleaner rebased version below:
https://github.com/kubernetes/kops/pull/6345
running the dependency commits only once.
